src/gen_prog_align.jl

```diff
@@ -0,0 +1,664 @@
+                    parent_objective_value = maximum([scores[parent_1], scores[parent_2]])
+                else
+                    parent_objective_value = scores[parent_1]
+                end
+               
+                # If the objective value of the child is bigger than the objective value of the parent
+                # and the child doesn't result in identical individuals in the new generation it is selected
+                # to be part of the next generation.
+                if new_objective_value > parent_objective_value && child ∉ new_population
+                    new_population[replacement_i] = child
+                    new_scores[replacement_i] = new_objective_value
+                    n_operations_without_improvement = 0
+                    n_operations +=1
+                    break
+                else 
+                    n_operations_without_improvement +=1
+                    n_operations +=1
+                    # Stop criterion
+                    if n_operations_without_improvement > max_operations_without_improvement || n_operations > max_operations
+                        
+                        go_on = false
+                        break
+                    end
+                end
+                #Update the renew array
+                if renew[operation[1]] == 10
+                    renew[operation[1]] = 1
+                else
+                    renew[operation[1]] += 1
+                end
+
+
+            end
+            
+        end
+        # If all replacements are found the next generation is set as the current generation, together with it's scores.
+        scores = deepcopy(new_scores)
+        population = deepcopy(new_population)
+    end 
+    # The best alignment, it's score and the sequence ID's are returned.
+    return population[sortperm(scores)[end]], maximum(scores), seq_ids
+
+end
+    
+     
+
+
+"""
+## fasta\\_to\\_array
+    fasta_to_array(path_to_fasta_file::String)
+
+Function that reads in a fasta file and returns 2 arrays. One array with the distinct sequences as strings, the other
+one with the corresponding sequence id's.
+"""
+function fasta_to_array(path_to_fasta_file::String)
+    fasta_file_handle = FASTA.Reader(open(path_to_fasta_file))
+    sequence_array = Array{String, 1}(undef, 0)
+    seq_ids = Array{String, 1}(undef, 0)
+
+    for a_record in fasta_file_handle
+        push!(sequence_array, sequence(a_record))
+        push!(seq_ids, FASTA.identifier(a_record))
+    end
+    close(fasta_file_handle)
+    return sequence_array, seq_ids
+end
+
+
+
+"""
+## Alignment
+    Alignment(seq_array; p = 0.95)
+
+Creates a random initial alignment starting from the sequence\\_array generated with fasta\\_to\\_array.
+Random gaps are inserted between the residues with a chance 1 - p.
+The sequence lengths are equalized. An array with the sequences as strings is returned.
+"""
+function Alignment(seq_array; p = 0.95)
```

Comment:
I'd call it `random_alignment` then. The best function name is the one which requires no additional explanation.